### PR TITLE
#54 adding Vscode debugger

### LINKED ISSUE #54: Adding the VScode debugging support
Adding support to debug the backend express code. The Backend during the development nodemon is watching for changes and restarting the express server. So adding the debugger would help during the local development. 

https://code.visualstudio.com/docs/nodejs/nodejs-debugging

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #54 

I am able to attach the debugger after nodemon restart with code changes. 

These are some good reference 
https://dev.to/alex_barashkov/how-to-debug-nodejs-in-a-docker-container-bhi
https://stackoverflow.com/questions/43898947/docker-node-app-always-crashes-on-file-change-using-nodemon/56832975#56832975

I tested this with code from master and LTIJS integration (without serverless approach). The Debugger is established on Nodemon so should effect with or without LTIJS

### CODE REVIEW

**Review by ssciolla — APPROVED** [review]
Looks like we have a conflict to resolve now, but I tried this briefly on Friday and it seemed to work for me. I don't have as much experience with remote debuggers though, so I'd trust @jonespm more on whether you can do what you need to do with it.

**Comment by pushyamig** [comment]
I resolved conflicts, so this branch is up to date from master.
@jonespm I really like your input on this

File: ccm_web/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "1.0.0",
   "description": "",
   "scripts": {
-    "dev:server": "nodemon --config nmon_config.json server/app.ts",
+    "dev:server": "nodemon --inspect=0.0.0.0 --config nmon_config.json server/app.ts",
```
Comment (jonespm) [inline ccm_web/package.json:6]: Is this where you'd put the port? I'd probably make this explicit like

-inspect=0.0.0.0:9229

**Comment by jonespm** [comment]
I should have time to try this tomorrow. Did it work as expected with the Typescript code? I had to use [ts-node ](https://www.npmjs.com/package/ts-node) and` "sourceMap": true` in the tsconfig.json in another project for the breakpoints and step-by-step to work with Typescript.

**Comment by pushyamig** [comment]
> I should have time to try this tomorrow. Did it work as expected with the Typescript code? I had to use ts-node and "sourceMap": true in the tsconfig.json in another project for the breakpoints and step-by-step to work with Typescript.

Yes, all files on server are Typescript and  was able to step in and over if debug is needed to code external package.json libraries

**Comment by pushyamig** [comment]
I am merging this.

**Pull request merged**